metadata/metadata.go

```diff
@@ -67,50 +67,56 @@ func (m *MetadataClient) GetMetadataDnsRecords() (map[string]utils.DnsRecord, er
 }
 
 func (m *MetadataClient) getContainersDnsRecords(dnsEntries map[string]utils.DnsRecord, serviceName string, stackName string) error {
-	containers, err := m.MetadataClient.GetContainers()
+	services, err := m.MetadataClient.GetServices()
 	if err != nil {
 		return err
 	}
 
 	ourFqdns := make(map[string]struct{})
-	for _, container := range containers {
-		if len(container.ServiceName) == 0 || len(container.Ports) == 0 || !containerStateOK(container) {
+	for _, service := range services {
+		if service.Kind != "service" {
 			continue
 		}
 
-		if len(serviceName) != 0 {
-			if serviceName != container.ServiceName {
+		for _, container := range service.Containers {
+			if len(container.Ports) == 0 || !containerStateOK(container) {
 				continue
 			}
-			if stackName != container.StackName {
-				continue
+
+			if len(serviceName) != 0 {
+				if serviceName != container.ServiceName {
```

Comment:
as we are iterating over service.Containers here, I think this block here (85-93) is irrelevant?